agent/Shooter.py

```diff
@@ -195,4 +221,35 @@ def get_hearts(self,
         for i in cards:
             if (38 < i <52):
                 hearts.append(i)
-        return hearts
\ No newline at end of file
+        return hearts
+
+    def points_broken(self,
+                      partial_state: HeartsState):
+        """Check to see if points have been broken."""
+        played_cards = np.where(partial_state.values >= 10)[0]
+        # Get the hearts cards (the where does not really matter, we just need a count of them)
+        points = np.where(played_cards > 38)[0]
+        if (len(points) > 0):
+            # At least one heart has been played so points are broken
+            return True
+        # Check if the queen of spades has been played
+        points = np.where(played_cards == 36)[0]
+        if (len(points) > 0):
+            # The queen was played
+            return True
+        # None of the points cards have been played
+        return False
+
+    def following_lead(self,
+                       partial_state: HeartsState,
+                       cards: list):
+        """Check if the player is capable of following suit before trying to take the trick.
+        Receives the partial state and a list representing the players hand."""
+        # Find the leading card to figure out its suit
+        lead = np.where(partial_state.values >= 30)[0][0]
+        # Divide by 13 to find the suit type
+        if ((cards[0]/13) == (lead/13)):
+            # Player has to follow if they have a card that is the same suit as the leader
+            return True
+        # Player can not follow the trick leader
+        return True
```

Comment:
Line 251 - 255: 
I don't know if this was intended, but returns True twice =)
Line 251 `if ((cards[0]/13) == (lead/13)):` might have an issue as this will never be true (cards[0] and lead can be in the same suit but have different results from division by 13). ie 16/13 != 18/13 but both are diamonds.
I would use math.floor. 
Replacement: `if math.floor(cards[0]/13) == math.floor(lead/13):`
Line 255  Return False.

Reply:
I forgot to even think about Python changing from an int to a float, thank you. Fixed.